### ISSUE TITLE
Proposal: New, decoupled SVG rendering

### ISSUE THREAD

**Issue by lukemcdo** [microsoft/microsoft-ui-xaml#3666]
# Proposal: Introduce a new, decoupled SVG rendering system to replace the UWP option

## Summary
UWP SVG rendering is relatively slow and lacks substantial features compared to many modern SVG renderers, so a change would be appreciated.

The first goal of this would simply be to no longer guarantee that SVGs render the same in dot releases, to allow for future work in replacing SVG rendering.

## Rationale
*  A new library for SVG rendering in WinUI could bring SVGs in WinUI/UWP apps to near-performance parity with PNGs, which brings a variety of upsides to applications that want to support HDPI displays without large file sizes.
* A new library could improve the cross-platform nature of rendering SVGs. This could improve consistency for Xamarin/MAUI/Uno.
* Decoupling this also continues the philosophical trend of WinUI 3: allow for out-of-band work to improve the ecosystem, and engage the community.

## Scope
| Capability | Priority |
| :---------- | :------- |
| This proposal will allow developers to use a performant, modern SVG renderer | Must |
| This proposal will allow developers to choose their own SVG rendering library | Could |
| This proposal will allow developers to choose the old UWP SVG rendering library | Could |
| This proposal will allow end users to use any DPI screen (SVG rerendering with multimonitor) | Should |

Worth noting -- I don't have a candidate library in mind. I just know that there are a lot of people itching for SVG in more places in WinUI, UWP, and throughout the Windows GUI, and the groundwork of decoupling the SVG rendering seems important toward this goal.

**Comment by michael-hawker** [comment]
🦙 I believe [Win2D itself has better support for SVG](http://microsoft.github.io/Win2D/html/T_Microsoft_Graphics_Canvas_Svg_CanvasSvgDocument.htm), there's a wrapper based on [Win2D here](https://github.com/mntone/SvgForXaml) and a [Skia based one here](https://github.com/wieslawsoltes/Svg.Skia) with it's own [SourceGenerator version here](https://github.com/wieslawsoltes/SourceGenerators) too.

I think it'd probably be best to just build on top of one of these solutions and if anything provide better hooks for dynamic XAML/Binding/Themes/Animation using them instead.

**Comment by jevansaks** [comment]
> 🦙 I believe [Win2D itself has better support for SVG](http://microsoft.github.io/Win2D/html/T_Microsoft_Graphics_Canvas_Svg_CanvasSvgDocument.htm),

I don't think so, Win2D just delegates the work to D2D which is what XAML uses too. See the link at the bottom of that doc to https://docs.microsoft.com/en-us/windows/win32/direct2d/svg-support?redirectedfrom=MSDN

**Comment by lmcdougald** [comment]
Even if it doesn't, I'm hopeful for some sort of notice in the WinUI 3 RTM release commenting that SVG output may not be identical. As I understand it, otherwise changing it could be considered a breaking change which couldn't go in a 3.X release.

**Comment by lukemcdo** [comment]
Why would you assume that the interfaces would change when the entire request is to change the underlying implementation? This issue is in no danger of reaching that scope and your comment doesn't feel like it belongs in a GitHub thread of this scope.

**Comment by HO-COOH** [comment]
I have encountered so many weird svg rendering issues where the result differs greatly from chrome. I guess the svg rendering pipeline has not updated since the very beginning of UWP and is continued in ~~winui 3~~ winui (updated branding now)

**Comment by HO-COOH** [comment]
With some help from AI, I successfully built skia for UWP with vcpkg by using patches for port files, and successfully rendered what UWP/win2d/winui3/SvgImageSource/CanvasSvgDocument are unable to render.

<img width="2362" height="1341" alt="Image" src="https://github.com/user-attachments/assets/62fd459b-9220-4661-b4a9-b81006e1ce3a" />

Compared to the original rendering:

https://github.com/microsoft/microsoft-ui-xaml/issues/11013

**Comment by mdtauk** [comment]
One thing I would like to see with any future support for improved SVG rendering, is the ability to apply brushes so SVGs could adopt light or dark theming, as well as accent colour support.

With Files we have had to develop a complicated control to layer various XAML paths in order to allow for themable vector icons, and being able to replace that with SVGs using xaml resource colours, would be a great simplification.

**Comment by castorix** [comment]
I had also tested [SVG.NET](https://github.com/svg-net/SVG), based on MS code, 
which can draw correctly some .SVG files that have problems in XAML or Direct2D

**Comment by HO-COOH** [comment]
Thanks for the resvg library, [I made a winui3 component from it](https://github.com/HO-COOH/WinUIEssentials#svgimagesource) and get much better result than what's built-in.
